metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/StellarPredicateProcessor.java

```diff
@@ -51,6 +48,11 @@ public Boolean parse( String rule
     if(rule == null || isEmpty(rule.trim())) {
       return true;
     }
-    return super.parse(rule, variableResolver, functionResolver, context);
+    try {
+      return super.parse(rule, variableResolver, functionResolver, context);
+    } catch (ClassCastException e) {
+      // predicate must return boolean
+      throw new IllegalArgumentException(String.format("The rule '%s' does not return a boolean value.", rule));
```

Comment:
Can you pass in `e` as an argument to the IAE so the user can tell the type that was returned.  Alternatively, you could add e.getMessage() to the IAE message.